pkg/gitops/profile.go

```diff
@@ -26,69 +28,134 @@ const (
 
 // Profile represents a gitops profile
 type Profile struct {
-	Processor fileprocessor.FileProcessor
-	Path      string
-	GitOpts   git.Options
-	GitCloner git.TmpCloner
-	FS        afero.Fs
-	IO        afero.Afero
-	clonedDir string
+	Processor         fileprocessor.FileProcessor
+	ProfileCloner     git.TmpCloner
+	UserRepoGitClient *git.Client
+	FS                afero.Fs
+	IO                afero.Afero
+}
+
+// Install installs the profile specified in the cluster config into a user's repository
+func (p *Profile) Install(clusterConfig *api.ClusterConfig) error {
+	if !clusterConfig.HasBootstrapProfile() {
+		return nil
+	}
+
+	userRepo := clusterConfig.Git.Repo
+
+	// Clone user's repo to apply Quick Start profile
+	gitCfg := clusterConfig.Git
+	bootstrapProfile := clusterConfig.Git.BootstrapProfile
+
+	// Clone user's repo to apply Quick Start profile
+	usersRepoName, err := git.RepoName(gitCfg.Repo.URL)
+	if err != nil {
+		return err
+	}
+	usersRepoDir, err := ioutil.TempDir("", usersRepoName+"-")
+	if err != nil {
+		return errors.Wrapf(err, "unable to create temporary directory for %q", usersRepoName)
+	}
+	bootstrapProfile.OutputPath = filepath.Join(usersRepoDir, "base")
+	logger.Debug("Directory %s will be used to clone the configuration repository and install the profile", usersRepoDir)
```

Comment:
I know this line was taken from elsewhere and you're aware of the convention for starting log messages with a lowercase letter, but just wanted to point it out:

```suggestion
	logger.Debug("directory %s will be used to clone the configuration repository and install the profile", usersRepoDir)
```